release-25.1: pkg/kv/kvpb: Add SafeFormat method to BatchResponse struct

Abhinav1299 opened the pull request:
Backport:
  * 1/1 commits from "pkg/kv/kvpb: Add SafeFormat method to BatchResponse struct" (#147609)
  * 1/1 commits from "kvclient: update error formatting in BatchResponse and improve test expectations in dist_sender_test" (#148570)

Please see individual PRs for details.

/cc @cockroachdb/release

----
This patch helps to fix the overly redacted log line on instances where BatchResponse is logged. Implemented a SafeFormat function on BatchResponse struct which overrides the String() method when its object is logged.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

----

Release justification: Resolves high priority overly redacted log line.

Abhinav1299 committed:
pkg/kv/kvpb: Add SafeFormat method to BatchResponse struct

This patch helps to fix the overly redacted log line on instances where BatchResponse
is logged. Implemented a SafeFormat function on BatchResponse struct which overrides
the String() method when its object is logged.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

Abhinav1299 committed:
kvclient: update error formatting in BatchResponse and improve test expectations in dist_sender_test

This commit modifies the error formatting in the BatchResponse's SafeFormat method to use a more concise representation. Additionally, it updates the expected output in the dist_sender_test to reflect changes in the string formatting, ensuring that the tests accurately validate the slow log messages.

No functional changes were made to the logic.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/148889)
<!-- Reviewable:end -->

Abhinav1299 committed:
kvclient: refine slow log message formatting in dist_sender_test

This update modifies the expected output format for slow log messages in the dist_sender_test to align with recent changes in error representation.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None
